test/Dapr.Client.Test/InvokeApiTest.cs

```diff
@@ -419,6 +421,39 @@ public async Task InvokeMethodAsync_WithReturnTypeAndData_WithNonDefaultVerb_Wit
             response.Name.Should().Be(invokedResponse.Name);
         }
 
+        [Fact]
+        public async Task InvokeMethodAsync_WithCancelledToken()
+        {
+            var client = new MockClient();
+            var data = new Response() { Name = "Look, I was invoked!" };
+            var invokeResponse = new InvokeResponse();
+            invokeResponse.Data = TypeConverters.ToAny(data);
+            
+            var response = 
+                client.Call<InvokeResponse>()
+                .SetResponse(invokeResponse)
+                .Build();
+
+            const string rpcExceptionMessage = "Call canceled by client";
+            const StatusCode rpcStatusCode = StatusCode.Cancelled;
+            const string rpcStatusDetail = "Call canceled";
+
+            var rpcStatus = new Status(rpcStatusCode, rpcStatusDetail);
+            var rpcException = new RpcException(rpcStatus, new Metadata(), rpcExceptionMessage);
+
+            // Setup the mock client to throw an Rpc Exception with the expected details info
+            client.Mock
+                .Setup(m => m.InvokeServiceAsync(It.IsAny<Autogen.Grpc.v1.InvokeServiceRequest>(), It.IsAny<CallOptions>()))
+                .Throws(rpcException);
+
+            var ctSource = new CancellationTokenSource();
+            CancellationToken ct = ctSource.Token;
+            ctSource.Cancel();
+            Request request = new Request() { RequestParameter = "Hello " };
+            var task = client.DaprClient.InvokeMethodAsync<Request, Response>("test", "test", request, cancellationToken: ct);
+            (await FluentActions.Awaiting(async () => await task).Should().ThrowAsync<OperationCanceledException>()).WithInnerException<Grpc.Core.RpcException>();
```

Comment:
You should move line 453 inside the lambda here. This code is a little bit fragile - calling a task returning method and then awaiting it later is a pretty expert thing to do. 

The reason is that `client.DaprClient.InvokeMethodAsync(...)` might throw an exception immediately, or it might throw when you await the task. You want them both be covered by the test.

Reply:
If needed we can jump on a call and I can explain.

Reply:
note: this is the the case for all of the new tests

Reply:
Thanks for taking the time to go over this with an example!